lib/iphone-interop-simulator-base.ts

```diff
@@ -0,0 +1,275 @@
+///<reference path="./.d.ts"/>
+"use strict";
+
+import child_process = require("child_process");
+import errors = require("./errors");
+import fs = require("fs");
+import Future = require("fibers/future");
+import options = require("./options");
+import os = require("os");
+import path = require("path");
+import util = require("util");
+import utils = require("./utils");
+
+var $ = require("NodObjC");
+
+export class IPhoneInteropSimulatorBase {
+	constructor(private simulator: IInteropSimulator) { }
+
+	private static FOUNDATION_FRAMEWORK_NAME = "Foundation";
+	private static APPKIT_FRAMEWORK_NAME = "AppKit";
+
+	private static DVT_FOUNDATION_RELATIVE_PATH = "../SharedFrameworks/DVTFoundation.framework";
+	private static DEV_TOOLS_FOUNDATION_RELATIVE_PATH = "../OtherFrameworks/DevToolsFoundation.framework";
+	private static CORE_SIMULATOR_RELATIVE_PATH = "Library/PrivateFrameworks/CoreSimulator.framework";
+	private static SIMULATOR_FRAMEWORK_RELATIVE_PATH_LEGACY = "Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks/DVTiPhoneSimulatorRemoteClient.framework";
+	private static SIMULATOR_FRAMEWORK_RELATIVE_PATH = "../SharedFrameworks/DVTiPhoneSimulatorRemoteClient.framework";
+
+	private static DEFAULT_TIMEOUT_IN_SECONDS = 90;
+
+	public run(appPath: string): IFuture<void> {
+		return this.execute(this.launch, { canRunMainLoop: true, appPath: appPath });
+	}
+
+	private launch(appPath: string): void {
+		var sessionDelegate = $.NSObject.extend("DTiPhoneSimulatorSessionDelegate");
```

Comment:
This function is very long and unreadable. Can we split it? I suggest to move out setting up the session delegate into a dedicated function.